keras_cv/models/regnet.py

```diff
@@ -0,0 +1,1546 @@
+        "widths": [48, 112, 256, 608],
+        "group_width": 16,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y008": {
+        "depths": [1, 3, 8, 2],
+        "widths": [64, 128, 320, 768],
+        "group_width": 16,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y016": {
+        "depths": [2, 6, 17, 2],
+        "widths": [48, 120, 336, 888],
+        "group_width": 24,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y032": {
+        "depths": [2, 5, 13, 1],
+        "widths": [72, 216, 576, 1512],
+        "group_width": 24,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y040": {
+        "depths": [2, 6, 12, 2],
+        "widths": [128, 192, 512, 1088],
+        "group_width": 64,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y064": {
+        "depths": [2, 7, 14, 2],
+        "widths": [144, 288, 576, 1296],
+        "group_width": 72,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y080": {
+        "depths": [2, 4, 10, 1],
+        "widths": [168, 448, 896, 2016],
+        "group_width": 56,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y120": {
+        "depths": [2, 5, 11, 1],
+        "widths": [224, 448, 896, 2240],
+        "group_width": 112,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y160": {
+        "depths": [2, 4, 11, 1],
+        "widths": [224, 448, 1232, 3024],
+        "group_width": 112,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y320": {
+        "depths": [2, 5, 12, 1],
+        "widths": [232, 696, 1392, 3712],
+        "group_width": 232,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+}
+
+BASE_DOCSTRING = """Instantiates the {name} architecture.
+
+  Reference:
+    - [Designing Network Design Spaces](https://arxiv.org/abs/2003.13678)
+    (CVPR 2020)
+
+  For image classification use cases, see [this page for detailed examples](https://keras.io/api/applications/#usage-examples-for-image-classification-models).
+
+  For transfer learning use cases, make sure to read the [guide to transfer learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+
+
+  The naming of models is as follows: `RegNet<block_type><flops>` where
+  `block_type` is one of `(X, Y)` and `flops` signifies hundred million
+  floating point operations. For example RegNetY064 corresponds to RegNet with
+  Y block and 6.4 giga flops (64 hundred million flops).
+
+  Args:
+    include_rescaling: whether or not to Rescale the inputs.If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+    include_top: Whether to include the fully-connected
+        layer at the top of the network.
+    classes: Optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
```

Comment:
`classes` should be specified iff `include_top` is true (independent of whether `weights` is specified)

Reply:
sgtm. If you can update this `classes` docstring, I'll go ahead and and merge this one

Reply:
It would be interesting to benchmark the basic trainer against the regnet trainer on these, if they're not too expensive to train. They seem to be following much the same idea, but we've been thinking about the idea of specializing trainers for architectures vs the idea of having a general script that gets them "good enough",  if I understood correctly?

Reply:
Done.